go-controller/pkg/factory/factory.go

```diff
@@ -1326,11 +1326,19 @@ func (wf *WatchFactory) GetEndpointSlice(namespace, name string) (*discovery.End
 	return endpointSliceLister.EndpointSlices(namespace).Get(name)
 }
 
-// GetServiceEndpointSlice returns the endpointSlice associated with a service
-func (wf *WatchFactory) GetEndpointSlices(namespace, svcName string) ([]*discovery.EndpointSlice, error) {
+// GetEndpointSlices returns the endpointSlices associated with a service for the specified network
+// if network is empty the default endpointSlices are returned, otherwise the function looks for mirror endpointslices
+// for the specified network.
+func (wf *WatchFactory) GetEndpointSlices(namespace, svcName, network string) ([]*discovery.EndpointSlice, error) {
```

Comment:
`GetEndpointSlices` is more abstract than `GetAllPodsBySelector` and I don't currently see any benefit in introducing `GetEndpointSlicesBySelector`. I think that it is ok to have the label selector abstracted away and rely on the network name+serviceName instead.

Reply:
Well quite the opposite `GetEndpointSlices` is too concrete since it's referencing not all the endpoint slices but the ones referencing a "concrete" label that points to the network.

I will not block on this but I think factory.go need to have more generic functions to help others and then we put at utils more concrete stuff.

Reply:
> Well quite the opposite GetEndpointSlices is too concrete since it's referencing not all the endpoint slices but the ones referencing a "concrete" label that points to the network.

Without my change the function already looks for a concrete label: `kubernetes.io/service-name`.

> I will not block on this but I think factory.go need to have more generic functions to help others and then we put at utils more concrete stuff.

I see your point and I think this makes sense. I will do that.

Reply:
Discussed offline, I renamed the `GetEndpointSlices` to `GetServiceEndpointSlices` and introduced the `GetEndpointSlicesBySelector` for any future use.